pulsar-common/src/main/java/org/apache/pulsar/common/events/EventsTopicNames.java

```diff
@@ -45,12 +45,6 @@
             Collections.unmodifiableSet(Sets.newHashSet(NAMESPACE_EVENTS_LOCAL_NAME, TRANSACTION_BUFFER_SNAPSHOT));
 
     public static boolean checkTopicIsEventsNames(TopicName topicName) {
-        String name;
-        if (topicName.isPartitioned()) {
-            name = TopicName.get(topicName.getPartitionedTopicName()).getLocalName();
-        } else {
-            name = topicName.getLocalName();
-        }
-        return EVENTS_TOPIC_NAMES.contains(name);
+        return EVENTS_TOPIC_NAMES.contains(TopicName.get(topicName.getPartitionedTopicName()).getLocalName());
```

Comment:
Please add a unit test to cover partitionedTopic and non-partitionedTopic

Reply:
`TopicName#getPartitionedTopicName` will check whether `topic` is partitioned and give the corresponding solution to handle these two situations. That’s why I did this simplification.